### PR TITLE
feature(rwx): add FilesystemResize method and request

### PR DESCRIPTION
**Pull request by james-munson**
#### Which issue(s) this PR fixes:

Issue:  https://github.com/longhorn/longhorn/issues/9736 aka https://github.com/longhorn/longhorn/issues/8118

#### What this PR does / why we need it:

Add an RPC call for FilesystemResize to the share-manager interface.

### CODE REVIEW

**Comment by derekbit** [comment]
@james-munson need to execute `make generate_grpc` to regenerate files

File: protobuf/smrpc/smrpc.proto [hunk 1]
```diff
@@ -10,6 +10,7 @@ option go_package = "github.com/longhorn/types/pkg/generated/smrpc";
 import "google/protobuf/empty.proto";
 
 service ShareManagerService {
+	rpc FilesystemResize(google.protobuf.Empty) returns (google.protobuf.Empty) {}
```
Comment (derekbit) [inline protobuf/smrpc/smrpc.proto:13]: No need the new and old sizes?

Comment (james-munson) [reply protobuf/smrpc/smrpc.proto:13]: That's correct - the fs utility, either `resize2fs` or `xfs_growfs` recognizes the necessary growth and just does the right thing.

**Review by PhanLe1010 — COMMENTED** [review]
In General, LGTM

File: protobuf/smrpc/smrpc.proto (hunk 1 above)
Comment (PhanLe1010) [reply protobuf/smrpc/smrpc.proto:13]: Do we need a method to get the current size of the filesystem inside share manager? Like this one is `set` method and the new one is `get` method?

Comment (james-munson) [reply protobuf/smrpc/smrpc.proto:13]: A good question.  Shall I write a separate ticket for that?  I'd hate to hold this up for it.

Comment (PhanLe1010) [reply protobuf/smrpc/smrpc.proto:13]: Thinking about it again, I think we can skip it for now. We can add them later when we need the new function

**Pull request merged**